### PR TITLE
chore: remove unnecessary use of username in webui (DET-6922)

### PR DESCRIPTION
**Pull request by johnkim-det**
## Description

Changing to use `userId` whenever appropriate.

usage of `username` should be limited to:
- authentication/login
- creating sample users for testing
- displaying all user info (`UserSettings` modal)
- `getDisplayName` util function
- resource allocation charts (to be handled separately in DET-7137)

all other usage of username should be replaced with `userId` and `getDisplayName` with `users` in the store.

## Test Plan

Could test everything that involves user data, or just make sure changes pass regression testing.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/hooks/useCreateModelModal.tsx [hunk 1]
```diff
@@ -80,7 +80,6 @@ const useCreateModelModal = (): ModalHooks => {
         labels: tags,
         metadata: metadata,
         name: modelName,
-        username: user?.username,
```
Comment (hkang1) [inline webui/react/src/hooks/useCreateModelModal.tsx:83]: **question:** just checking to see if is ok to create a model without a user id. Does it figure it out on the backend based on the auth token of who is creating a model?

File: webui/react/src/hooks/useTelemetry.ts [hunk 2]
```diff
@@ -82,7 +82,7 @@ class Telemetry {
           clusterName: info.clusterName,
           edition: 'OSS',
           masterId: info.masterId,
-          username: auth.user.username,
+          userId: auth.user.id,
```
Comment (hkang1) [inline webui/react/src/hooks/useTelemetry.ts:85]: **chore:** for segment traits, the user ID trait key is `id` according to [this](https://segment.com/docs/connections/spec/identify/#traits)

File: webui/react/src/utils/user.ts [hunk 3]
```diff
@@ -1,5 +1,10 @@
 import { DetailedUser, User } from 'types';
 
-export function getDisplayName (user: DetailedUser | User | undefined): string {
+interface UserNameFields {
+  displayName?: string;
+  username: string;
```
Comment (hkang1) [inline webui/react/src/utils/user.ts:5]: **question:** should `username` be optional?

**Review by hkang1 — APPROVED** [review]
LGTM! I have a few questions, posted, but nothing seems to be blocking, one chore somewhere in there.

File: webui/react/src/hooks/useCreateModelModal.tsx (hunk 1 above)
Comment (johnkim-det) [reply webui/react/src/hooks/useCreateModelModal.tsx:83]: yep, it checks `CurrentUser`

File: webui/react/src/hooks/useTelemetry.ts (hunk 2 above)
Comment (johnkim-det) [reply webui/react/src/hooks/useTelemetry.ts:85]: reverting changes to this file

File: webui/react/src/utils/user.ts (hunk 3 above)
Comment (johnkim-det) [reply webui/react/src/utils/user.ts:5]: yep missed this, thanks

**Pull request merged**